climada/hazard/base.py

```diff
@@ -369,6 +377,415 @@ def set_vector(self, *args, **kwargs):
+        an error due to missing data or faulty data types.
+
+        Notes
+        -----
+        * Intensity and fraction data must be three-dimensional data that is interpreted
+          with the coordinates given by ``coordinate_vars`` (or the default). All other
+          data must be given in time coordinates only.
+        * To avoid confusion in the call signature, several parameters are keyword-only
+          arguments.
+        * The attributes ``Hazard.tag.haz_type`` and ``Hazard.unit`` currently cannot be
+          read from the Dataset. Use the method parameters to set these attributes.
+        * This method does not read coordinate system metadata. Use the ``crs`` parameter
+          to set a custom coordinate system identifier.
+        * This method **does not** read lazily. Single data arrays must fit into memory.
+
+        Parameters
+        ----------
+        data : xarray.Dataset or str
+            The data to read from. May be a opened dataset or a path to a raster data
+            file, in which case the file is opened first. Works with any file format
+            supported by ``xarray``.
+        hazard_type : str
+            The type identifier of the hazard. Will be stored directly in the hazard
+            object.
+        intensity_unit : str
+            The physical units of the intensity. Will be stored in the ``hazard.tag``.
+        intensity : str, optional
+            Identifier of the `xarray.DataArray` containing the hazard intensity data.
+        coordinate_vars : dict(str, str), optional
+            Mapping from default coordinate names to coordinate names used in the data
+            to read. The default is
+            ``dict(event="time", longitude="longitude", latitude="latitude")``
+        data_vars : dict(str, str), optional
+            Mapping from default variable names to variable names used in the data
+            to read. The default names are ``fraction``, ``hazard_type``, ``frequency``,
+            ``event_name``, and ``event_id``. If these values are not set, the method
+            tries to load data from the default names. If this fails, the method uses
+            default values for each entry. If the values are set to empty strings
+            (``""``), no data is loaded and the default values are used exclusively. See
+            examples for details.
+
+            Default values are:
+            * ``fraction``: 1.0 where intensity is not zero, else zero
+            * ``hazard_type``: Empty string
+            * ``frequency``: 1.0 for every event
+            * ``event_name``: String representation of the event time
+            * ``event_id``: Consecutive integers starting at 1 and increasing with time
+        crs : str, optional
+            Identifier for the coordinate reference system to use. Defaults to
+            ``EPSG:4326`` (WGS 84), defined by ``climada.util.constants.DEF_CRS``.
```

Comment:
@chahank I implemented the `crs` argument in ccec87f8, but I'm unsure how to test this argument properly. Do you have an idea or suggestion?

Reply:
Great addition!

What do you mean exactly by testing? Is it about how to automatically retrieve it from the netcdf file or is it about how to write a unit test on that specifically?